### PR TITLE
update gradle build to compile against the Dart plugin

### PR DESCRIPTION
**Pull request by devoncarew**
- update gradle build to compile against the Dart plugin
- inline DartTestUtils.java
- compile against the current stable version of IntelliJ CE + the stable version of the Dart plugin
- tests currently disabled on travis pending some debugging

@pq @stevemessick

### CODE REVIEW

**Comment by devoncarew** [comment]
This is very close to working now - just one more issue. The testSrc classes from the Dart plugin may not be visible to the Flutter plugin?

```
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:prepareSandbox UP-TO-DATE
:buildPlugin UP-TO-DATE
:jar UP-TO-DATE
:assemble UP-TO-DATE
:processTestResources UP-TO-DATE
:compileTestJava
/Users/devoncarew/projects/intellij/flutter-intellij/testSrc/io/flutter/FlutterCodeInsightFixtureTestCase.java:9: error: cannot find symbol
import com.jetbrains.lang.dart.util.DartTestUtils;
                                   ^
  symbol:   class DartTestUtils
  location: package com.jetbrains.lang.dart.util
/Users/devoncarew/projects/intellij/flutter-intellij/testSrc/io/flutter/FlutterCodeInsightFixtureTestCase.java:24: error: cannot find symbol
      DartTestUtils.configureDartSdk(myModule, getTestRootDisposable(), true);
      ^
  symbol:   variable DartTestUtils
  location: class FlutterCodeInsightFixtureTestCase
2 errors
:compileTestJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

**Comment by devoncarew** [comment]
(https://travis-ci.org/flutter/flutter-intellij/builds/157749443#L306)

**Comment by pq** [comment]
This is awesome.

LGTM 👍

File: src/io/flutter/run/FlutterRunner.java [hunk 1]
```diff
@@ -41,7 +41,9 @@ protected DartUrlResolver getDartUrlResolver(@NotNull final Project project, @No
     return new FlutterUrlResolver(project, contextFileOrDir);
   }
 
-  @Override
+  // TODO(devoncarew): This override is commented out until the next stable rev.
```
Comment (stevemessick) [inline src/io/flutter/run/FlutterRunner.java]: This should be OK. Alex merged the change to super.

File: testSrc/io/flutter/util/DartTestUtils.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import com.jetbrains.lang.dart.sdk.DartSdkUtil;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Assert;
+
+import java.io.File;
+import java.util.Collections;
+
+// TODO(devoncarew): We should pare this class down, and move some of its functionality into FlutterTestUtils.
+// From //intellij-plugins/Dart/testSrc/com/jetbrains/lang/dart/util/DartTestUtils.java
+public class DartTestUtils {
```
Comment (stevemessick) [inline testSrc/io/flutter/util/DartTestUtils.java]: This should be in the com.jetbrains.lang.dart package with the other copied classes (which I'm about to commit).

**Comment by stevemessick** [comment]
`testSrc` would have to be exported from the Dart plugin to be visible to Flutter, I think.

File: testSrc/io/flutter/util/DartTestUtils.java (hunk 2 above)
Comment (devoncarew) [inline testSrc/io/flutter/util/DartTestUtils.java]: I'll rebase my PR and move this class.

File: src/io/flutter/run/FlutterRunner.java (hunk 1 above)
Comment (devoncarew) [inline src/io/flutter/run/FlutterRunner.java]: Alex just released a new Dart plugin w/ this change today - I'll update to that version in the build script.

At runtime, we may want to check for a min. required version of the Dart plugin.

Comment (devoncarew) [inline src/io/flutter/run/FlutterRunner.java]: (removed this TODO: and added the override back in)

**Comment by devoncarew** [comment]
> testSrc would have to be exported from the Dart plugin to be visible to Flutter, I think.

Not sure why I couldn't see it from the gradle build. It is a slightly different process then when the plugin src is loaded in IntelliJ - it downloads the plugin binary and compiles against the class files.

**Comment by stevemessick** [comment]
I don't think test sources are distributed with the plugin. It would
surprise me if they were, anyway.

File: testSrc/io/flutter/util/DartTestUtils.java (hunk 2 above)
Comment (devoncarew) [inline testSrc/io/flutter/util/DartTestUtils.java]: I'm having issues w/ classpaths w/ this; I'll update the third_party/intellij-plugins-dart readme and plan to refactor in a follow-up PR.

Comment (stevemessick) [inline testSrc/io/flutter/util/DartTestUtils.java]: Did you create `testSrc` at the same level as `src` (which I just added) and mark it as a `test source` directory? That's in the *.iml files. See the lines for the existing `testSrc` and `src` for examples.

**Pull request merged**